### PR TITLE
Adds a hash of environment variables to the cronjob.

### PR DESCRIPTION
**Pull request by yeled**
{ 'a' => 'b'}, becomes a=b before the cronjob line

### CODE REVIEW

File: manifests/d.pp [hunk 1]
```diff
@@ -54,6 +54,9 @@
 # ranges (e.g. 0-10), and ranges with steps (e.g. 0-10/2). An asterisk (*)
 # is equivalent to 0-59, and also supports steps (e.g. */20).
 #
+# [*comment*]
```
Comment (keymone) [inline manifests/d.pp]: should be \*env\* ?

File: templates/d.erb [hunk 2]
```diff
@@ -9,6 +9,11 @@ PATH=/usr/local/bin:/usr/local/sbin:/nail/sys/bin:/usr/sbin:/usr/bin:/sbin:/bin
 <% else -%>
 PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/bin:/usr/local/sbin
 <% end -%>
+<% if ! @env.empty? -%>
```
Comment (keymone) [inline templates/d.erb]: this is redundant, `.each` on empty hash will yield nothing

File: templates/d.erb [hunk 3]
```diff
@@ -9,6 +9,11 @@ PATH=/usr/local/bin:/usr/local/sbin:/nail/sys/bin:/usr/sbin:/usr/bin:/sbin:/bin
...
+<% @env.each do |variable, value| %>
+<%= variable %>=<%= value -%>
```
Comment (keymone) [inline templates/d.erb]: wrapping `value` in `"quotes"` would be useful here

File: templates/d.erb [hunk 4]
```diff
@@ -9,6 +9,11 @@ PATH=/usr/local/bin:/usr/local/sbin:/nail/sys/bin:/usr/sbin:/usr/bin:/sbin:/bin
 <% else -%>
 PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/bin:/usr/local/sbin
 <% end -%>
+<% if ! @env.empty? -%>
+<% @env.each do |variable, value| %>
```
Comment (keymone) [inline templates/d.erb]: order of `.each` over a map is non-deterministic, use this instead:

    <% @env.keys.sort.each do |variable| value=@env[variable] %>

Comment (keymone) [inline templates/d.erb]: btw this will also fix tests which are failing because ruby 1.8 doesn't support `.each` over hashmap

**Comment by solarkennedy** [comment]
Just junk 1.8.7 in travis.yml. It stopped getting support 3 years ago.

File: templates/d.erb (hunk 4 above)
Comment (yeled) [inline templates/d.erb]: how does this then extract `value` from the || ?

File: templates/d.erb (hunk 3 above)
Comment (yeled) [inline templates/d.erb]: yes, good idea.

File: templates/d.erb (hunk 4 above)
Comment (yeled) [inline templates/d.erb]: i trolled myself with naming things `value`. fixed now.

**Comment by keymone** [comment]
:ship:

**Comment by solarkennedy** [comment]
Great job! Ship!

**Pull request merged**